homeassistant/helpers/translation.py

```diff
@@ -528,6 +529,38 @@ def async_translations_loaded(hass: HomeAssistant, components: set[str]) -> bool
     )
 
 
+@callback
+def async_get_exception_message(
+    translation_domain: str | None,
+    translation_key: str | None,
+    translation_placeholders: dict[str, str] | None = None,
+) -> str | None:
+    """Return a translated exception message.
+
+    Defaults to English, requires translations to already be cached.
+    """
+    language = "en"
+    if translation_domain is None or translation_key is None:
+        return None
+
+    hass = async_get_hass()
+    localize_key = (
+        f"component.{translation_domain}.exceptions.{translation_key}.message"
+    )
+    translations = async_get_cached_translations(hass, language, "exceptions")
+    if localize_key in translations:
+        if message := translations[localize_key]:
+            message = message.rstrip(".")
+        if not translation_placeholders:
+            return message
+        with suppress(KeyError):
+            message = message.format(**translation_placeholders)
+        return message
+
+    # We return the translation key when was not found in the cache
+    return localize_key
```

Comment:
I think we should return the translation key, to improve readability.
For example, I think `not_valid_fan_mode` is more readable than `component.climate.exceptions.not_valid_fan_mode.message`

```suggestion
    return translation_key
```